src/pytorch_lightning/accelerators/hpu.py

```diff
@@ -52,13 +55,27 @@ def get_parallel_devices(devices: int) -> List[torch.device]:
 
     @staticmethod
     def auto_device_count() -> int:
-        """Get the devices when set to auto."""
-        # TODO(@kaushikb11): Update this when api is exposed by the Habana team
-        return 8
+        """Returns the number of HPU devices when the devices is set to auto."""
+        try:
+            return torch_hpu.device_count()
```

Comment:
What versions have this?

Reply:
With synapse release 1.5, we have this interface enabled

Reply:
This above logic would be better:

```python
if _HPU_VERSION_X:
	return torch_hpu.device_count()
return 8
```

Reply:
I see your point but this is just for sanity